api/engineimage.go

```diff
@@ -6,21 +6,30 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/pkg/errors"
 	"github.com/rancher/go-rancher/api"
+	"github.com/rancher/go-rancher/client"
 )
 
-func (s *Server) EngineImageList(rw http.ResponseWriter, req *http.Request) (err error) {
+func (s *Server) EngineImageList(rw http.ResponseWriter, req *http.Request) error {
 	apiContext := api.GetApiContext(req)
 
+	eil, err := s.engineImageList(nil)
+	if err != nil {
+		return err
+	}
+	apiContext.Write(eil)
+	return nil
+}
+
+func (s *Server) engineImageList(apiContext *api.ApiContext) (*client.GenericCollection, error) {
```

Comment:
Anyone using those `apiContext`?

Reply:
Not in this impl, but the method signature is purposely `listFunc func(ctx *api.ApiContext) (*client.GenericCollection, error)`. Check out `api/stream.go:NewStreamHandlerFunc`. The request context is used by Go Rancher API `github.com/rancher/go-rancher` in other resource listers to create URIs for actions.

Reply:
I see. In that case, let's pass the apiContext instead of `nil` to the function.